### PR TITLE
8344337: SecurityManager cleanup in java.prefs module

### PR DESCRIPTION
**Pull request by bchristi-git**
Remove usages of SecurityManager, doPrivildged, and AccessController from the java.prefs module.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344337](https://bugs.openjdk.org/browse/JDK-8344337): SecurityManager cleanup in java.prefs module (**Bug** - P4)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22252/head:pull/22252` \
`$ git checkout pull/22252`

Update a local copy of the PR: \
`$ git checkout pull/22252` \
`$ git pull https://git.openjdk.org/jdk.git pull/22252/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22252`

View PR using the GUI difftool: \
`$ git pr show -t 22252`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22252.diff">https://git.openjdk.org/jdk/pull/22252.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22252#issuecomment-2486507273)
</details>

### CODE REVIEW

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 1]
```diff
@@ -27,9 +27,9 @@
 
 import java.util.*;
 import java.io.*;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
-import java.security.PrivilegedExceptionAction;
+//import java.security.AccessController;
+//import java.security.PrivilegedAction;
+//import java.security.PrivilegedExceptionAction;
 import java.security.PrivilegedActionException;
```
Comment (LanceAndersen) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:33]: I think you can remove these vs leaving them commented out

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 2]
```diff
@@ -53,22 +53,21 @@ class FileSystemPreferences extends AbstractPreferences {
...
-        PrivilegedAction<Void> load = () -> {
+//        PrivilegedAction<Void> load = () -> {
             System.loadLibrary("prefs");
-            return null;
-        };
-        AccessController.doPrivileged(load);
+//            return null;
+//        };
+//        AccessController.doPrivileged(load);
     }
```
Comment (LanceAndersen) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:63]: Looks like there is some clean up here to do of the commented out code...

**Review by LanceAndersen — CHANGES_REQUESTED** [review]
Hi Brent,

There looks to be a lot of commented out code that should be removed as part of your PR?

**Comment by seanjmullan** [comment]
A few classes need updated copyrights.

File: src/java.prefs/share/classes/java/util/prefs/Preferences.java [hunk 3]
```diff
@@ -267,11 +245,7 @@ public String run() {
                 }
             }
         }
-
-        return AccessController.doPrivileged(
-            new PrivilegedAction<PreferencesFactory>() {
-                public PreferencesFactory run() {
-                    return factory1();}});
+        return factory1();
```
Comment (RogerRiggs) [inline src/java.prefs/share/classes/java/util/prefs/Preferences.java:248]: No need to make factory1 a separate method, merge into factory.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 4]
```diff
@@ -765,56 +735,35 @@ public synchronized void sync() throws BackingStoreException {
...
-                new PrivilegedExceptionAction<Void>() {
-                public Void run() throws BackingStoreException {
-                    syncSpiPrivileged();
-                    return null;
-                }
-            });
-        } catch (PrivilegedActionException e) {
-            throw (BackingStoreException) e.getException();
-        }
+        syncSpiPrivileged();
```
Comment (RogerRiggs) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:766]: Inline syncSpiPrivileged(), it doesn't need to be separate and the name "privileged" is misleading

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java (hunk 2 above)
Comment (bchristi-git) [reply src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:63]: Whoops!

File: src/java.prefs/share/classes/java/util/prefs/Preferences.java [hunk 5]
```diff
@@ -30,20 +30,10 @@
...
 import java.io.IOException;
 import java.io.OutputStream;
-import java.security.AccessController;
-import java.security.Permission;
-import java.security.PrivilegedAction;
 import java.util.Iterator;
 import java.util.ServiceLoader;
 import java.util.ServiceConfigurationError;
 
-// These imports needed only as a workaround for a JavaDoc bug
```
Comment (bchristi-git) [inline src/java.prefs/share/classes/java/util/prefs/Preferences.java:40]: This comment was added quite a while ago. But FWIW, a specdiff comparison shows no JavaDoc differences.

**Comment by bchristi-git** [comment]
> I think the try-block at line 137 in FileSystemPreferences would be more readable if it were indented. I know it was not indented in the original, either.

Oh, yeah, I don't like that either. :(
Fixed; also added {}s.
Thanks.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 6]
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
...
-                    getLogger().warning(e.toString());
+                    getLogger().warning("Could not change permissions" +
+                        " on userRoot directory. ");
                 }
-                userRootModTime = userRootModFile.lastModified();
-                return null;
+                getLogger().info("Created user preferences directory.");
             }
-        });
+            else
```
Comment (bplb) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:127]: It seems odd that `else` is not on line 125 after the bracket. If putting it on the same line makes it  less readable, then maybe add a pair of brackets for the `else`.

**Review by bplb — APPROVED** [review]
Looks all right to me now, but you might want also to get a second opinion.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 7]
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
...
+                getLogger().info("Created user preferences directory.");
+            } else {
+                getLogger().warning("Couldn't create user preferences" +
+                        " directory. User preferences are unusable.");
             }
-        });
+        }
+        isUserRootWritable = userRootDir.canWrite();
+        String USER_NAME = System.getProperty("user.name");
+        userLockFile = new File (userRootDir,".user.lock." + USER_NAME);
```
Comment (turbanoff) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:133]:
```suggestion
        userLockFile = new File(userRootDir, ".user.lock." + USER_NAME);
```

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 8]
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
...
-                                                          USER_RWX_ALL_RX);
-                            } catch (IOException e) {
-                            }
-                        } else {
-                            getLogger().warning("Could not create "
-                                + "system preferences directory. System "
-                                + "preferences are unusable.");
-                        }
+        String systemPrefsDirName =
+          System.getProperty("java.util.prefs.systemRoot","/etc/.java");
```
Comment (turbanoff) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:177]:
```suggestion
          System.getProperty("java.util.prefs.systemRoot", "/etc/.java");
```

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 9]
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
...
-                }
-                systemRootModTime = systemRootModFile.lastModified();
-                return null;
             }
-        });
+        }
+        isSystemRootWritable = systemRootDir.canWrite();
+        systemLockFile = new File(systemRootDir, ".system.lock");
+        systemRootModFile =
+                       new File (systemRootDir,".systemRootModFile");
```
Comment (turbanoff) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:207]:
```suggestion
                       new File(systemRootDir, ".systemRootModFile");
```

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 10]
```diff
@@ -596,43 +566,32 @@ private void initCacheIfNecessary() {
...
-                    return null;
-                }
-            });
-        } catch (PrivilegedActionException e) {
-            throw (BackingStoreException) e.getException();
+            newLastSyncTime = prefsFile.lastModified();
+            try (FileInputStream fis = new FileInputStream(prefsFile)) {
+                XmlSupport.importMap(fis, m);
+            }
+        } catch(Exception e) {
```
Comment (turbanoff) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:577]:
```suggestion
        } catch (Exception e) {
```

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 11]
```diff
@@ -185,58 +172,53 @@ static Preferences getSystemRoot() {
...
-                systemRootModTime = systemRootModFile.lastModified();
-                return null;
             }
-        });
+        }
+        isSystemRootWritable = systemRootDir.canWrite();
+        systemLockFile = new File(systemRootDir, ".system.lock");
+        systemRootModFile =
+                       new File (systemRootDir,".systemRootModFile");
+        if (!systemRootModFile.exists() && isSystemRootWritable)
```
Comment (turbanoff) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:208]: let's add braces here too

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 12]
```diff
@@ -117,52 +109,47 @@ static Preferences getUserRoot() {
...
-                }
-                isUserRootWritable = userRootDir.canWrite();
-                String USER_NAME = System.getProperty("user.name");
-                userLockFile = new File (userRootDir,".user.lock." + USER_NAME);
-                userRootModFile = new File (userRootDir,
-                                               ".userRootModFile." + USER_NAME);
-                if (!userRootModFile.exists())
+        userRootDir =
+              new File(System.getProperty("java.util.prefs.userRoot",
+              System.getProperty("user.home")), ".java/.userPrefs");
```
Comment (RogerRiggs) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:115]: This expression would be more readable if the 2nd line was indented.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java [hunk 13]
```diff
@@ -466,16 +447,11 @@ public void run() {
...
-                Runtime.getRuntime().addShutdownHook(
-                    new Thread(null, null, "Sync Timer Thread", 0, false) {
-                    public void run() {
-                        syncTimer.cancel();
-                        syncWorld();
-                    }
-                });
-                return null;
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(null, null, "Sync Timer Thread", 0, false) {
```
Comment (RogerRiggs) [inline src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:451]: Future work, use a Virtual thread for this.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java (hunk 10 above)
Comment (bchristi-git) [reply src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:577]: Most catch statements in this file do not have a space.

File: src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java (hunk 13 above)
Comment (bchristi-git) [reply src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java:452]: https://bugs.openjdk.org/browse/JDK-8344793

**Review by RogerRiggs — APPROVED** [review]
Looks good

**Review by LanceAndersen — APPROVED** [review]
Latest updates look good

**Review by bplb — APPROVED** [review]
+1

File: src/java.prefs/macosx/classes/java/util/prefs/MacOSXPreferencesFile.java [hunk 14]
```diff
@@ -82,15 +82,9 @@ class MacOSXPreferencesFile {
...
+    @SuppressWarnings("restricted")
     private static void loadPrefsLib() {
-        java.security.AccessController.doPrivileged(
-            new java.security.PrivilegedAction<Void>() {
-                public Void run() {
-                    System.loadLibrary("prefs");
-                    return null;
-                }
-            });
+        System.loadLibrary("prefs");
```
Comment (irisclark) [inline src/java.prefs/macosx/classes/java/util/prefs/MacOSXPreferencesFile.java:87]: What am I missing?  Why keep the private static method if it only contains a single method invocation and is only used in the static initializer?

Comment (bchristi-git) [reply src/java.prefs/macosx/classes/java/util/prefs/MacOSXPreferencesFile.java:87]: We want to `@SuppressWarnings("restricted")` the `loadLibrary()` call. IIUC, I can't `@SuppressWarnings` the static block, nor the call to `loadLibrary()`, so AFAICT that call needs to be in a separate, `@SuppressWarnings`-ed method.

Comment (seanjmullan) [reply src/java.prefs/macosx/classes/java/util/prefs/MacOSXPreferencesFile.java:87]: Right. The other option is to put the annotation above the class definition which would cover the entire class, but I think this is better as it makes the annotation specific to that method only.

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**